Extensions/Firebase/A_utils/A_UIDArray.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Firebase Tools Collection
+ * @fileoverview
+ * @author arthuro555
+ */
+
+/**
+ * A special array where push tries to reuse old unused indices.
```

Comment:
Is this really giving any upside compared to a map (JS object), or an object based on a map?
I'm a bit afraid of these kind of micro-optimizations that make the codebase more complex and not really 100% indispensable - a few uids in memory in a JS object won't kill anyone? :)

Reply:
Note that in the case of heavily used arrays of game objects, this could be useful - with appropriate benchmarks showing that this is giving an advantage. Here I think we don't need any extra complexity ;)

Reply:
It's not just UIDs, if I don't delete the snapshot it won't be garbage collected and it's a bigger object. Since the default array has no remove action, I did that to add it and also gain some performance.
Not deleting the snapshots or replacing them with null would never completely release the memory they took and would result in a memory leak, which is very bad.

Reply:
Sounds are indeed a very heavy resource (in the sense that too much of them will crash the game/browser) that we need to destroy/recycle. They also don't have ids, so we use an array and check if a sound is finished, and if yes we erase it by the new one, to be sure we keep the minimum number of sounds in memory.

For an upload snapshot, would rather a more traditional JS object associating ids to snapshot work? 
Memory consumption will be equivalent, but access will be in constant time (`O(1)`) while using an array, access is linear in the number of elements stored in the array. You have to think how to reuse ids. You can either keep an array for that, or just don't reuse id and increase them at every usage - potentielly avoiding mistakes for people using this in GDevelop :) 

**I am also fine using your approach using an array** but in this case, could you do an object that does not "inherit" from Array? I say that because otherwise this means an external user can mess with your internal array, doing splice, concat, etc... Potentially using the array in a way you don't want to, and creating bugs. When designing a class, make the public interface as limited as possible. 
As we say, [always favor composition over inheritance](https://en.wikipedia.org/wiki/Composition_over_inheritance#:~:text=Composition%20over%20inheritance%20(or%20composite,than%20inheritance%20from%20a%20base) :) With a limited interface with only the needed functions, you ensure that your class is "foolproof" and won't be misused.